eodag/plugins/download/http.py

```diff
@@ -1087,11 +1093,14 @@ def get_chunks(stream: Response) -> Any:
                     f"Local asset detected. Download skipped for {asset['href']}"
                 )
                 continue
-
+            if matching_url and re.match(matching_url, asset["href"]):
+                auth_object = auth
+            else:
+                auth_object = None
```

Comment:
if `matching_conf` is used instead of `matching_url` (see [auth plugins doc](https://eodag.readthedocs.io/en/latest/plugins_reference/auth.html)), `auth_object` will be `None`.
You'd better always authenticate in this case

Reply:
changed